libr/util/sys.c

```diff
@@ -411,20 +439,19 @@ R_API int r_sys_crash_handler(const char *cmd) {
 
 	free (crash_handler_cmd);
 	crash_handler_cmd = strdup (cmd);
-	sigact.sa_handler = signal_handler;
-	sigemptyset (&sigact.sa_mask);
-	sigact.sa_flags = 0;
-	sigaddset (&sigact.sa_mask, SIGINT);
-	sigaddset (&sigact.sa_mask, SIGSEGV);
-	sigaddset (&sigact.sa_mask, SIGBUS);
-	sigaddset (&sigact.sa_mask, SIGQUIT);
-	sigaddset (&sigact.sa_mask, SIGHUP);
-
-	sigaction (SIGINT, &sigact, (struct sigaction *)NULL);
-	sigaction (SIGSEGV, &sigact, (struct sigaction *)NULL);
-	sigaction (SIGBUS, &sigact, (struct sigaction *)NULL);
-	sigaction (SIGQUIT, &sigact, (struct sigaction *)NULL);
-	sigaction (SIGHUP, &sigact, (struct sigaction *)NULL);
+
+	sigemptyset (&mask);
+	sigaddset (&mask, SIGINT);
+	sigaddset (&mask, SIGSEGV);
+	sigaddset (&mask, SIGBUS);
+	sigaddset (&mask, SIGQUIT);
+	sigaddset (&mask, SIGHUP);
+
+	r_sys_sigaction (SIGINT, signal_handler, mask);
+	r_sys_sigaction (SIGSEGV, signal_handler, mask);
```

Comment:
as long as we are setting the same signal handler to all of them aybe we can just have a asingle function that takes an list of signals and the handler. like this:

```
r_sys_sigaction (signal_handler, SIGINT, SIGSEGV,SIGBUS,SIGQUIT,SIGHUP, -1);
```

this single function can hide the sigset_t need to be public/wrapped so unless im missing an usecase for this i think this will be cleaner

Reply:
Yes this seems to be a good idea. But what do you think of using an array instead of va_args ?
```
r_sys_sigaction (signal_handler, (int []){SIGINT, SIGSEGV, SIGBUS, SIGQUIT, SIGHUP, -1});
```